config/v2_3_experimental/types/unit.go

```diff
@@ -120,12 +141,30 @@ func (d NetworkdDropin) Validate() report.Report {
 	return r
 }
 
-func validateUnitContent(content string) error {
+func validateUnitContent(content string) ([]*unit.UnitOption, error) {
 	c := strings.NewReader(content)
-	_, err := unit.Deserialize(c)
+	opts, err := unit.Deserialize(c)
 	if err != nil {
-		return fmt.Errorf("invalid unit content: %s", err)
+		return nil, fmt.Errorf("invalid unit content: %s", err)
+	}
+	return opts, nil
+}
+
+func (u Unit) validateInstallSection(contentSections []*unit.UnitOption) error {
```

Comment:
Since the validation framework looks for `Validate<field name>` functions, glancing through this code could trip someone up. Want to drop a comment here saying that this is intentionally not for the validator to call?

Reply:
Also since the error from this function is to be used as a warning, how about having it return a `report.Report`? That way the error type can be wrapped into a report entry marking it as a warning here, and not require the caller to interpret the results. This could then be called like: `r.Merge(validateInstallSection(opts))`.

Reply:
Also as I continue to nit-pick this specific function, maaaybe move it under `config/errors`, have it take an additional parameter of `enabled bool`, and have the caller have to do `u.Enable || ...`. That way there aren't four separate implementations of this.

I'd be fine with it staying here too though. Your call.

Reply:
Since this function is an un-exported `validate`, not `Validate`, I didn't feel a comment was necessary, happy to add it though.

Reply:
One of my fears in the Ignition codebase is I'll accidentally name something `validateFoo` instead of `ValidateFoo` and the validation for that field just won't happen, so the comment may be for my own benefit more than everyone else's.